R/ggMarginalHelpers.R

```diff
@@ -71,24 +60,74 @@ needsFlip <- function(marg, type) {
   topAndBoxP || rightAndNonBoxP
 }
 
-margPlotNoGeom <- function(marg, type, data) {
+margPlotNoGeom <- function(data, type, scatPbuilt, marginGroupColour,
+                           marginGroupFill) {
+  
+  mapping <- ggplot2::aes(x = var)
+  
+  haveMargMap <- marginGroupColour || marginGroupFill
+  
+  if (haveMargMap) {
 
-  # Build plot (sans geom). Note: Boxplot is the only plot type that needs y aes
-  if (type %in% c("boxplot", "violin")) {
-    plot <- ggplot2::ggplot(data = data, 
-                            ggplot2::aes_string(x = 'var', y = 'var'))
-  } else {
-    plot <- ggplot2::ggplot(data = data, ggplot2::aes_string(x = 'var'))
+    data <- data[, c("var", "colour"), drop = FALSE]
+    if (marginGroupFill) {
+      data[, "fill"] <- data[, "colour"]
+    }
+    
+    values <- unique(data$colour)
+    names(values) <- values
+    
+    if (marginGroupColour && !marginGroupFill) {
+      xtraMapNames <- "colour"
+    } else if (marginGroupColour && marginGroupFill) {
+      xtraMapNames <- c("colour", "fill")
+    } else {
+      xtraMapNames <- "fill"
+    }
+
+    xtraMap <- sapply(xtraMapNames, as.symbol, USE.NAMES = TRUE, 
+                      simplify = FALSE)
+    mapping <- structure(c(mapping, xtraMap), class = "uneval")
   }
 
-  if (needsFlip(marg = marg, type = type)) {
-    plot <- plot +  ggplot2::coord_flip()
+  # Boxplot is the only plot type that needs y aes
```

Comment:
this comment ("only") doesn't match the code (boxplot+violin) :)

Reply:
oh yeah i noticed that a while back and forgot to change. mind if i fix this in the re-styling pr?

Reply:
sure

Reply:
nvm, i gotta rebase again anyway so i might as well do it here.